base/linalg/arnoldi.jl

```diff
@@ -106,3 +106,51 @@ function eigs(A, B;
                                  resid, ncv, v, ldv, sigma, iparam, ipntr, workd, workl, lworkl, rwork)
 
 end
+
+
+## svds
+
+type SVDOperator{T,S} <: AbstractArray{T, 2}
+    X::S
+    m::Int
+    n::Int
+    SVDOperator(X::S) = new(X, size(X,1), size(X,2))
+end
+
+## v = [ left_singular_vector; right_singular_vector ]
+*{T,S}(s::SVDOperator{T,S}, v::Vector{T}) = [s.X * v[s.m+1:end]; s.X' * v[1:s.m]]
+size(s::SVDOperator)  = s.m + s.n, s.m + s.n
+issym(s::SVDOperator) = true
+
+function getOperatorType(X)
+  et = eltype(X)
+  if et <: Integer || et <: Float64
+    return Float64
+  end
+  if et <: Complex64
+    return Complex64
+  end
+  if et <: Complex
+    return Complex128
+  end
+  if et <: Float32
+    return Float32
+  end
+  error("Element type $et is not supported for 'svds'.")
+end
+
+function svds{S}(X::S; nsv::Int = 6, ritzvec::Bool = true, tol::Float64 = 0.0, maxiter::Int = 1000)
+  otype = getOperatorType(X)
```

Comment:
I think we can just simplify this to use just `eltype(X)` and avoid `getOperatorType()` altogether. The checks for whether eltype() is BlasFloat should ideally be in `eigs`. It should be possible to make `SVDOperator` free of types then.